### PR TITLE
[TDP-17] Checkpoints

### PR DESCRIPTION
**Pull request by funkychicken493**
**DEPENDS ON TDP-16 BEING ACCEPTED**

Adds in a checkpoint scene that can be reused indefinitely in a level. When the player dies without a checkpoint activated they will respawn at a predetermined spot in the level.

### CODE REVIEW

**Review by abby1332 — APPROVED** [review]
Checkpoint kinda creepy but pop off

**Review by ASlightlyOvergrownCactus — APPROVED** [review]
works and is creepy

File: player.gd [hunk 1]
```diff
@@ -72,10 +73,13 @@ var time_sliding: float = 0.0
...
 	death_particles.hide()
 	death_particles.emitting = false
-	spawn_point.teleport(self)
+	if point != null:
+		point.spawn_point.teleport(self)
+	else:
+		spawn_point.teleport(self)
 	animation_manager.show()
 	death_text.hide()
 	dead = false
```
Comment (ASlightlyOvergrownCactus) [inline player.gd:85]: add at end of code, make sure to set velocity to 0 after respawning!
`velocity.x = 0.0`
`velocity.y = 0.0`

**Review by ASlightlyOvergrownCactus — CHANGES_REQUESTED** [review]

File: player.gd (hunk 1 above)
Comment (funkychicken493) [reply player.gd:85]: teehee

**Review by ASlightlyOvergrownCactus — APPROVED** [review]
Alr it works now 👍

**Pull request merged**